Add documentation for removing auto-provisioned toolchains (#26519)
gradle/gradle: subprojects/docs/src/docs/userguide/jvm/toolchains.adoc

* Either start gradle using `-Porg.gradle.java.installations.auto-download=false`
* Or put `org.gradle.java.installations.auto-download=false` into a `gradle.properties` file.

[[sub:remove_auto_provisioned_toolchain]]
=== How to remove an auto-provisioned toolchain

> To properly remove an auto-provisioned toolchain, you can remove or modify any reference to the toolchain from `settings.gradle.kts` or `settings.gradle`

Reviewer: ```suggestion
To properly remove an auto-provisioned toolchain, you can remove or modify any reference to the toolchain from `settings.gradle.kts` or `settings.gradle`.
```

* Either start gradle using `-Porg.gradle.java.installations.auto-download=false`
* Or put `org.gradle.java.installations.auto-download=false` into a `gradle.properties` file.

[[sub:removing_auto_provisioned_toolchain]]
=== Removing an auto-provisioned toolchain

There are cases where a user might want to take explicit control over the toolchain configuration for their project. This could include specifying custom versions of a toolchain, ensuring compatibility with specific project requirements, or addressing issues related to the auto-provisioned toolchain.

> In a situation where it's necessary to remove an auto-provisioned toolchain, remove the relevant toolchain located in the `/jdks` directory within the <<directory_layout.adoc#dir:gradle_user_home,Gradle User Home>>.

Reviewer: ```suggestion
When removing an auto-provisioned toolchain is necessary, remove the relevant toolchain located in the `/jdks` directory within the <<directory_layout.adoc#dir:gradle_user_home,Gradle User Home>>.
```

* Either start gradle using `-Porg.gradle.java.installations.auto-download=false`
* Or put `org.gradle.java.installations.auto-download=false` into a `gradle.properties` file.

[[sub:removing_auto_provisioned_toolchain]]
=== Removing an auto-provisioned toolchain

> In some cases, a user might want explicit control over the toolchain configuration for their project.

Reviewer 2: This line and the next one do not provide any new information.
The section is in the toolchains chapter so there is no need to re-iterate what toolchains provide.

Please remove this line and the next one.
